single_kernel_mongo/managers/ldap.py

```diff
@@ -90,53 +91,61 @@ def on_ldap_ready(self, relation: Relation) -> None:
             raise WaitingForLdapDataError("Waiting for LDAP data.")
 
         if not ldap_data.ldaps_urls:
-            logger.error("You must configure Glauth with LDAPS support")
+            logger.error("You must configure Glauth with LDAPS (ldaps_enabled=true) support")
             raise LDAPSNotEnabledError("LDAPS not enabled.")
 
         if self.charm.unit.is_leader():
             self.state.ldap.set_from(ldap_data)
-
-        self.restart_when_ready()
+            # ops event to restart if ready for the leader.
+            # The other units will restart during the ldap-peers-relation-changed hook.
+            self.dependent.ldap_events.restart_if_ready_event.emit()
 
     def restart_when_ready(self):
         """Restarts when we are ready."""
-        if not self.state.ldap_relation:
-            logger.info(f"Not restarting, missing the `{ExternalRequirerRelations.LDAP}` relation")
-            return
-        if not self.state.ldap_cert_relation:
-            logger.info(
-                f"Not restarting, missing the `{ExternalRequirerRelations.LDAP_CERT}` relation"
-            )
-            return
-
-        if self.state.ldap.is_ready():
-            logger.info("Restarting mongodb server for LDAP integration")
-            self.dependent.restart_charm_services()
-            self.charm.status_manager.set_and_share_status(ActiveStatus())
+        match self.get_status():
+            case None:
+                return
+            case ActiveStatus():
+                logger.info("Restarting mongodb server for LDAP integration")
+                self.dependent.restart_charm_services()
+                self.charm.status_manager.set_and_share_status(ActiveStatus())
+            case status:
+                self.charm.status_manager.set_and_share_status(status)
 
     def on_ldap_unavailable(self) -> None:
         """Runs when the LDAP integration is broken."""
         self.state.ldap.clean_databag()
 
         self.dependent.restart_charm_services()
+        self.charm.status_manager.set_and_share_status(self.get_status() or ActiveStatus())
```

Comment:
Is the relation already unavailable when the ldap_unavailable event comes through? Because I wonder how `get_status` would behave in this context

Reply:
It's emitted in the `relation_broken` event.
`get_status` handles both events so it should be fine and if I'm not wrong, it should update the status so that we know there is only one of the relations integrated. I'll give it a try !

Reply:
it behaves as expected